src/core/Akka.Cluster.Tests/ClusterSpec.cs

```diff
@@ -306,6 +308,140 @@ public void A_cluster_must_be_allowed_to_join_and_leave_with_local_address()
             }
         }
 
+        [Fact]
+        public void A_cluster_must_be_able_to_JoinAsync()
```

Comment:
@Horusiath need a spec here that tests for "what happens if this method is called when we're already connected to the cluster?"

Reply:
It will immediately complete with success (I've seen this in the impl), but I guess you're right. We probably should have tests for that.